main.go

```diff
@@ -162,36 +230,60 @@ func run() error {
 		return fmt.Errorf("the repository is dirty: commit all changes before running")
 	}
 
-	err = w.Reset(&git.ResetOptions{Commit: *prev, Mode: git.HardReset})
-	if err != nil {
-		return fmt.Errorf("failed to reset the worktree to a previous commit: %w", err)
+	resetAndRunBenchmark := func(commit plumbing.Hash, ref string, isTag bool) (benchSet Set, err error) {
+		err = w.Reset(&git.ResetOptions{Commit: commit, Mode: git.HardReset})
+		if err != nil {
+			return nil, fmt.Errorf("failed to reset the worktree to a commit %v, ref %v: %w", commit, ref, err)
+		}
+
+		klog.InfoS("Run Benchmark", "commitHash", commit, "Ref", ref)
+		var tagVersion string
+		if isTag {
+			tagVersion = ref
+		}
+		benchSet, err = runBenchmarks(tagVersion)
+		if err != nil {
+			return nil, fmt.Errorf("failed to run a benchmark: %w", err)
+		}
+		return
 	}
 
 	defer func() {
 		_ = w.Reset(&git.ResetOptions{Commit: head.Hash(), Mode: git.HardReset})
 	}()
-
 	updateBenchmarks()
 
-	klog.InfoS("Run Benchmark", "prev", prev, "baseRef", baseRef)
-	prevSet, err := runBenchmarks()
+	// run benchmark of baseRef
+	prevSet, err := resetAndRunBenchmark(*prev, baseRef, false)
 	if err != nil {
-		return fmt.Errorf("failed to run a benchmark: %w", err)
+		return err
 	}
 
-	err = w.Reset(&git.ResetOptions{Commit: head.Hash(), Mode: git.HardReset})
-	if err != nil {
-		return fmt.Errorf("failed to reset the worktree to HEAD: %w", err)
+	// run benchmark of latestReleaseVersion
+	var latestReleaseSet Set
+	var tagName string
+	var prevVersionTag *plumbing.Reference
+	if compareLatestVersion {
+		prevVersionTag, err = getLatestRelease(r)
+		if err != nil {
+			return fmt.Errorf("failed to get latest release version: %w", err)
+		}
+		tagName = prevVersionTag.Name().String()
+		latestReleaseSet, err = resetAndRunBenchmark(prevVersionTag.Hash(), prevVersionTag.Name().Short(), true)
+		if err != nil {
+			klog.ErrorS(err, "Failed to run a benchmark")
```

Comment:
I'm curious: why do we log the error here, instead of returning an error, like we do in other places?

Reply:
I think we can see the logs of benchmark results compared with baseRef even run the benchmark failed compared with the latest version.